### ISSUE TITLE
nothing visible in the main panel

### ISSUE THREAD

**Issue by mazzoo** [hitchhikr/protrekkr#11]
![Screenshot_2024-02-08_10-07-44](https://github.com/hitchhikr/protrekkr/assets/151651/eb8e0f38-f6da-4328-8877-fc52ec033149)

just built under arch linux, nothing to see in the main (track) panel.

There's no output on the console about missing fonts whatsoever. the scrollbars nicely seem to follow the active line while playing.

**Comment by hitchhikr** [comment]
You've got the file "pattern_font.bmp" in the skins directory ?

**Comment by mazzoo** [comment]
have the font, also dsyfunct in full screen.

`% sha256sum release/distrib/skins/pattern_font.bmp 
d7213c3a674df490524b478012848e0c007e3f6ae2d2258be6d1bf2f6aedc6b3  release/distrib/skins/pattern_font.bmp`

I had just built using `make -j -f makefile.linux` and then I ran `release/distrib/ptk_linux` .
now I tried `cd release/distrib && ./ptk_linux` but it's the same.

**Comment by hitchhikr** [comment]
Try to download the repository again i changed the rendering to software surface, maybe that's the problem.

Frankly, i'm even surprised that it works on Linux, i tried on the latest Debian and it crashes randomly, i think it's a problem with the implementation of SDL, i plan to migrate the program to SDL2 hopefully it'll fix the many troubles i had with SDL1.

**Comment by mazzoo** [comment]
still not working in 5abb69ff333f11921c0e60fea0d9fa15b17be557
(also not in FS)

**Comment by hitchhikr** [comment]
Btw, are you using a 32 or 64 bit Distribution ?

**Comment by mazzoo** [comment]
64

**Comment by hitchhikr** [comment]
Try it again, it should work now...

**Comment by mazzoo** [comment]
I am sorry, but no, it doesn't, am @ b80014ee0bf288c0c320cf47cab1f216d604ef32

**Comment by hitchhikr** [comment]
It works properly on debian so i don't know what's wrong with your linux.

**Comment by hitchhikr** [comment]
Say; you wouldn't be using sdl12-compat by any chance ?

**Comment by mazzoo** [comment]
```
% ldd ./release/distrib/ptk_linux | grep -i sdl
	libSDL-1.2.so.0 => /usr/lib/libSDL-1.2.so.0 (0x00007a7b2e463000)
```
pacman says `extra/sdl12-compat 1.2.68-1 [installed]` but also `extra/sdl2 2.30.0-1 [installed: 2.28.5-1]`

oh then the build seems to have picked the ancient SDL version?

**Comment by hitchhikr** [comment]
I tried with sdl12-compat on mac os and indeed it didn't show the patterns.

**Comment by SamuraiCrow** [comment]
I looked at the "required by" entry for sdl12-compat package on Manjaro Linux and SDL-Draw wasn't listed.

> [samuraicrow@manjaro-pc ~]$ pamac info sdl12-compat
> Name                  : sdl12-compat
> Version               : 1.2.68-1
> Description           : SDL 1.2 runtime compatibility library using SDL 2.0
> URL                   : https://github.com/libsdl-org/sdl12-compat
> Licenses              : MIT
> Repository            : extra
> Installed Size        : 667.8 kB
> Groups                : --
> Depends On            : sdl2 glu
> Optional Dependencies : --
> Required By           : lib32-sdl12-compat libvisual sdl_gfx sdl_image
>                         sdl_mixer sdl_net sdl_sound sdl_ttf
>                         steam-native-runtime virtualbox wxgtk2.8
> Optional For          : grub libde265 libmpeg2 mlt vlc xine-lib
> Provides              : sdl=1:1.2.15+r406+gf1caf909-1
> Replaces              : sdl<=1:1.2.15+r406+gf1caf909
> Conflicts With        : sdl
> Packager              : Jelle van der Waa <jelle@archlinux.org>
> Build Date            : Wed 27 Sep 2023 09:08:37 AM CDT
> Install Date          : Wed 04 Oct 2023 04:54:01 AM CDT
> Install Reason        : Installed as a dependency for another package
> Validated By          : Signature
> Backup files          : --
> 
> [samuraicrow@manjaro-pc ~]$

**Comment by hitchhikr** [comment]
Try it with latest commit, it *may* work.

**Comment by mazzoo** [comment]
desole, non.

**Comment by hitchhikr** [comment]
remove sdl12-compat then.

**Comment by mazzoo** [comment]
you may not like it. when I remove `sdl12-compat` then the SDL headers are not found anymore. that is b/c SDL headers are kept in `/usr/include/SDL` and SDL2 headers are in `/usr/include/SDL2`

so I quickly changed the source to see (attachted patch) but then I get missing symbols like `SDLK_LAST` as per the attached make.log

[path_SDL_to_SDL2.diff.zip](https://github.com/hitchhikr/protrekkr/files/14335979/path_SDL_to_SDL2.diff.zip)

[make.log](https://github.com/hitchhikr/protrekkr/files/14335976/make.log)

**Comment by hitchhikr** [comment]
Well install libsdl1.2-dev ?

**Comment by hitchhikr** [comment]
Also it looks like sdm12-compat doesn't handle the right mouse button (at least it doesn't on mac os).

**Comment by mazzoo** [comment]
> Well install libsdl1.2-dev ?

nope, that's not the debian way on arch. a package comes with it's headers.
but it seems arch does not provide SDL1.2 anymore. only the SDL compat layer.

**Comment by hitchhikr** [comment]
You know the deal then: https://github.com/libsdl-org/SDL-1.2

**Comment by mazzoo** [comment]
wonderful. that works.

so the future way for Protrekkr will be SDL2? Or do you intend to fix the SDL12-compat layer still? the ancient SDL1.2 doesn't seem a way into the distributions anymore...

**Comment by hitchhikr** [comment]
I'll come to that eventually, but the problems are that some platforms i support do not have SDL 2 (i haven't investigated that much yet) and migrating from 1 to 2 isn't really that trivial, a lot of code need to be rewritten. I think i'll switch to OpenGL rendering in  a near future.

**Comment by SamuraiCrow** [comment]
I see that SDL_GFX has a compatibility shim for SDL2_GFX but SDL_Draw has no such shim. Do you think that's the actual culprit?

**Comment by hitchhikr** [comment]
The patterns are drawn with SDL_BlitSurface()

**Comment by theorize999** [comment]
I think I'm having a similar problem here. Built the SDL version posted above... at least I'm pretty sure it worked out. 
when I list SDL packages it returns:
libsdl-image1.2/noble,now 1.2.12-13build4 amd64 [installed,automatic]
libsdl-ttf2.0-0/noble,now 2.0.11-6 amd64 [installed,automatic]
libsdl1.2-dev/noble,now 1.2.68-2 amd64 [installed]
libsdl1.2debian/noble,now 1.2.68-2 amd64 [installed]
libsdl2-2.0-0/noble-updates,now 2.30.0+dfsg-1ubuntu3.1 amd64 [installed,automatic]
libsdl2-dev/noble-updates,now 2.30.0+dfsg-1ubuntu3.1 amd64 [installed,automatic]
sfxr-sdl/now 1.2.1-1 amd64 [installed,local]

Ubuntu 24.04.3 LTS

Getting some weird interface parts missing and it's been a while but it seems like certian interface elements aren't responding, like if I zaaap all I can't switch patttern or add sequencer positions, if I load a song it'll play but I can't do any of that or even clear the sequencer most of that does not work. I can move through song position. Synth controls work if switched on. Sample editor works. 303 controls work. Pattern does not show. Looks like I have 2 pattern fonts in skins and one general font as far as bitmaps go. when folding down bottom bar it seems like a bit of the interface remains on the left.

<img width="1920" height="996" alt="Image" src="https://github.com/user-attachments/assets/5a1b5338-55d3-42f1-94a7-1e1632789d39" />

<img width="1920" height="996" alt="Image" src="https://github.com/user-attachments/assets/ca8483d5-4b8b-45dd-8128-757f9b046eb0" />

<img width="1920" height="996" alt="Image" src="https://github.com/user-attachments/assets/26346c5c-b379-4941-bf92-f0b5fcc257e8" />
